scripts/bert/run_pretraining.py

```diff
@@ -125,119 +214,167 @@ def train(data_train, model, nsp_loss, mlm_loss, vocab_size, ctx, store):
 
     train_begin_time = time.time()
     begin_time = time.time()
-    running_mlm_loss = running_nsp_loss = running_num_tks = 0
+    running_mlm_loss, running_nsp_loss = 0, 0
+    running_num_tks = 0
     batch_num = 0
     step_num = args.start_step
 
-    parallel_model = ParallelBERT(model, mlm_loss, nsp_loss, vocab_size,
-                                  store.num_workers * accumulate, trainer=fp16_trainer)
-    num_ctxes = len(ctx)
+    logging.debug('Training started')
+
+    # create dummy data loader if needed
+    if args.dummy_data_len:
+        target_shape = (args.batch_size, args.dummy_data_len)
+        data_train = get_dummy_dataloader(data_train, target_shape)
+
+    parallel_model = DataParallelBERT(model, trainer=fp16_trainer)
+    num_ctxes = len(ctxs)
     parallel = nlp.utils.Parallel(num_ctxes if num_ctxes > 1 else 0, parallel_model)
 
     while step_num < num_train_steps:
-        for _, dataloader in enumerate(data_train):
+        for _, data_batch in enumerate(data_train):
```

Comment:
Is this correct? where is the `for _, data_batch in enumerate(dataloader):` iteration?

Reply:
the intermediate data iterator is now removed